RecoVertex/NuclearInteractionProducer/src/NuclearLikelihood.cc

```diff
@@ -4,15 +4,16 @@ void NuclearLikelihood::calculate(const reco::Vertex& vtx) {
   likelihood_ = 0.0;
   if (vtx.isValid()) {
     if (vtx.tracksSize() > 1) {
-      likelihood_ = 0.3;
-      int idBest;
+      int idBest = 0;
       int secMaxHits = secondaryTrackMaxHits(vtx, idBest);
-      if (secMaxHits > 3)
-        likelihood_ = 0.5;
-      if (secMaxHits > 4)
-        likelihood_ = 0.7;
       if ((*(vtx.tracks_begin() + idBest))->normalizedChi2() < 3.0)
```

Comment:
While editing this code, I was surprised of this line (that I did not modify): if there are no secondary tracks in the vertex (i.e. secMaxHits=0) and the chi2 of the main track is below 3 then likelihood_ is equal 1.

Could anybody confirm that this the wanted behaviour?
@mtosi

Reply:
I tried to grep a bit, and I did not find any active use cases of this code.

Reply:
Indeed...
- "NuclearLikelihood" is used inside ```RecoVertex/NuclearInteractionProducer/plugins/NuclearInteractionEDProducer.cc```
- ... whose plugin is used in ```RecoVertex/NuclearInteractionProducer/python/NuclearInteraction_cfi.py```
- ... which is imported by ```RecoParticleFlow/PFTracking/python/nuclear_cff.py```
- ... entering in a sequence defined in ```RecoParticleFlow/​PFTracking/​python/​nuclearRemainingHits_cff.py```
- ... and that file is never accessed in CMSSW

All those modules/configs were never updated since 2013 (at least), excluding a few technical adaptations to the developing framework from time to time. I suspect that they can all be erased, in particular now that some PF code cleanup is ongoing: @hatakeyamak @bendavid